common/variant/src/main/java/org/apache/spark/types/variant/VariantUtil.java

```diff
@@ -355,9 +380,12 @@ public static boolean getBoolean(byte[] value, int pos) {
 
   // Get a long value from variant value `value[pos...]`.
   // It is only legal to call it if `getType` returns one of `Type.LONG/DATE/TIMESTAMP/
-  // TIMESTAMP_NTZ`. If the type is `DATE`, the return value is guaranteed to fit into an int and
-  // represents the number of days from the Unix epoch. If the type is `TIMESTAMP/TIMESTAMP_NTZ`,
-  // the return value represents the number of microseconds from the Unix epoch.
+  // TIMESTAMP_NTZ/YEAR_MONTH_INTERVAL/DAY_TIME_INTERVAL`. If the type is `DATE`, the return value
+  // is guaranteed to fit into an int and represents the number of days from the Unix epoch.
+  // If the type is `TIMESTAMP/TIMESTAMP_NTZ`, the return value represents the number of
+  // microseconds from the Unix epoch. If the type is `YEAR_MONTH_INTERVAL`, the return value
```

Comment:
We should also mention that the year-month one is guaranteed to fit into an int.